channel.go

```diff
@@ -437,6 +437,32 @@ func (ch *Channel) UnBanUser(targetID string, options map[string]string) error {
 	return ch.client.UnBanUser(targetID, options)
 }
 
+// ShadowBan shadow bans userID from this channel
+// bannedByID: user who shadow bans userID.
+// options: additional shadow ban options, ie {"timeout": 3600, "reason": "offensive language is not allowed here"}.
+func (ch *Channel) ShadowBan(userID, bannedByID string, options map[string]interface{}) error {
```

Comment:
should we type the options since it only accepts certain fields?

Reply:
not sure, makes it less flexible. we use interface maps everywhere

Reply:
@yaziine what do you think?